### PR TITLE
Adapted changes from https://github.com/jc-m/libnetwork/tree/IPAliases

### PR DESCRIPTION
**Pull request by axqx**
Adapting changes proposed in https://github.com/jc-m/libnetwork/tree/IPAliases to support adding IP aliases to network interfaces inside containers.

### CODE REVIEW

File: drivers/remote/driver_test.go [hunk 1]
```diff
@@ -298,6 +326,7 @@ func TestRemoteDriver(t *testing.T) {
 		address:        "192.168.5.7/16",
 		addressIPv6:    "2001:DB8::5:7/48",
 		macAddress:     "ab:cd:ef:ee:ee:ee",
+		ipAliases:      []string{"192.168.255.255/32", "192.168.1.1/32"},
```
Comment (mauri) [inline drivers/remote/driver_test.go:329]: use 10.1.1.1/24 and 10.2.2.2/24 instead?

File: drivers/remote/driver_test.go [hunk 2]
```diff
@@ -527,6 +558,10 @@ func (r *rollbackEndpoint) AddressIPv6() *net.IPNet {
 	return nil
 }
 
+func (r *rollbackEndpoint) IPAliases() []*net.IPNet {
+	return nil
```
Comment (mauri) [inline drivers/remote/driver_test.go:562]: need to implement?

Comment (axqx) [reply drivers/remote/driver_test.go:562]: This is just a dummy method, but this functionality it is actually implemented and tested in the other test case.

File: drivers/remote/driver_test.go (hunk 1 above)
Comment (axqx) [inline drivers/remote/driver_test.go]: will change in future

**Comment by mauri** [comment]
LGTM!

**Pull request merged**